group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -526,6 +539,27 @@ public Assignment targetAssignment(String memberId) {
         return targetAssignment.getOrDefault(memberId, Assignment.EMPTY);
     }
 
+    /**
+     * Returns the target assignment of the member or the target assignment
+     * associated to the instance id if it does not exist.
```

Comment:
I feel like there is something off with this comment 😅 
Seems like we return the target assignment of the member ID if the instance ID doesn't exist.  And if we do have the instanceID we return the target assignment of the previousMemberId if it exists.